specification/resources/resource-manager/Microsoft.Resources/stable/2023-08-01/deploymentScripts.json

```diff
@@ -0,0 +1,971 @@
+        },
+        "retentionInterval": {
+          "type": "string",
+          "format": "duration",
+          "description": "Interval for which the service retains the script resource after it reaches a terminal state. Resource will be deleted when this duration expires. Duration is based on ISO 8601 pattern (for example P1D means one day)."
+        },
+        "timeout": {
+          "type": "string",
+          "format": "duration",
+          "default": "P1D",
+          "description": "Maximum allowed script execution time specified in ISO 8601 format. Default value is P1D"
+        }
+      }
+    },
+    "DeploymentScriptPropertiesBase": {
+      "type": "object",
+      "x-ms-external": true,
+      "description": "Common properties for the deployment script.",
+      "properties": {
+        "containerSettings": {
+          "type": "object",
+          "description": "Container settings.",
+          "$ref": "#/definitions/ContainerConfiguration"
+        },
+        "storageAccountSettings": {
+          "type": "object",
+          "description": "Storage Account settings.",
+          "$ref": "#/definitions/StorageAccountConfiguration"
+        },
+        "cleanupPreference": {
+          "type": "string",
+          "description": "The clean up preference when the script execution gets in a terminal state. Default setting is 'Always'.",
+          "enum": [
+            "Always",
+            "OnSuccess",
+            "OnExpiration"
+          ],
+          "default": "Always",
+          "x-ms-enum": {
+            "name": "cleanupOptions",
+            "modelAsString": true
+          }
+        },
+        "provisioningState": {
+          "type": "string",
+          "readOnly": true,
+          "description": "State of the script execution. This only appears in the response.",
+          "enum": [
+            "Creating",
+            "ProvisioningResources",
+            "Running",
+            "Succeeded",
+            "Failed",
+            "Canceled"
+          ],
+          "x-ms-enum": {
+            "name": "ScriptProvisioningState",
+            "modelAsString": true
+          }
+        },
+        "status": {
+          "type": "object",
+          "readOnly": true,
+          "description": "Contains the results of script execution.",
+          "$ref": "#/definitions/ScriptStatus"
+        },
+        "outputs": {
+          "description": "List of script outputs.",
+          "type": "object",
+          "readOnly": true,
+          "additionalProperties": {
+            "type": "object",
+            "description": "Script output in <name, value> pair."
+          }
+        }
+      }
+    },
+    "ContainerGroupSubnetId": {
+      "description": "Container group subnet information.",
+      "type": "object",
+      "properties": {
+        "id": {
```

Comment:
Is it an azure resource ID? If so, please add "format": "arm-id" and "x-ms-arm-id-details".

https://github.com/Azure/autorest/blob/main/docs/extensions/readme.md#x-ms-arm-id-details

Reply:
its an ARM resource ID. I made the changes as per the documentation